OpenTrons/opentrons-api
# feat(api): allow ungrip gripper labware while door is open

# Overview

closes [EXEC-734](https://opentrons.atlassian.net/browse/EXEC-734).
add ungrip command and allow queuing it and executing it while door is open.

## Test Plan and Hands on Testing

1. upload a protocol that will enter ER mode.
2. open door.
3. issue an ungrip command:
```
{
    "data": {
        "commandType": "unsafe/ungripLabware",
        "intent": "fixit",
        "params": {
        }
    }
}
```
4. make sure the gripper opens its jaw.

tested with dev server and the command succeeded but still need to test with an actual gripper (@SyntaxColoring thank you Max)

## Changelog

1. added an ungrip command `unsafe/ungripLabware`
2. added logic for queuing while the door is open.
3. added logic for allowing to execute while door is open.

## Review requests

1. gripper command makes sense?
2. logic changes make sense? 

## Risk assessment

medium. added a new command but need to make sure nothing has changed with the door saftey.


[EXEC-734]: https://opentrons.atlassian.net/browse/EXEC-734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
